src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs

```diff
@@ -434,6 +436,18 @@ protected ConnectionConfiguration(IConnectionPool connectionPool, IConnection co
 		public T BasicAuthentication(string username, SecureString password) =>
 			Assign(new BasicAuthenticationCredentials(username, password), (a, v) => a._basicAuthCredentials = v);
 
+		/// <summary>
+		/// Api Key to send with all requests to Elasticsearch
+		/// </summary>
+		public T ApiKeyAuthentication(string id, SecureString apiKey) =>
+			Assign(new ApiKeyAuthenticationCredentials(id, apiKey), (a, v) => a._apiKeyAuthentication = v);
+
+		/// <summary>
+		/// Api Key to send with all requests to Elasticsearch
+		/// </summary>
+		public T ApiKeyAuthentication(string id, string apiKey) =>

```

Comment:
Kept string password for BasicAuthentication in 7.x to ease the move from 6.x to 7.x, but wonder if we should we only support `SecureString` for apikey going forward?

Reply:
Would prefer to keep supporting `string` for ease of use.